warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeFactory.java

```diff
@@ -108,142 +112,151 @@ public static JexlNode createNodeTreeFromFieldsToValues(ContainerType containerT
                 allValues.addAll(values);
             }
             fieldsToValues.clear();
-            for (String identifier : JexlASTHelper.getIdentifierNames(orgNode)) {
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
                 fieldsToValues.put(identifier, allValues);
             }
         }
         
-        Set<String> fields = fieldsToValues.keySet();
+        List<JexlNode> children = new LinkedList<>();
         
-        JexlNode parentNode = (containerType.equals(ContainerType.OR_NODE) ? new ASTOrNode(ParserTreeConstants.JJTORNODE) : new ASTAndNode(
-                        ParserTreeConstants.JJTANDNODE));
-        int parentNodeChildCount = 0;
+        Set<String> fields = fieldsToValues.keySet();
         
         if (keepOriginalNode) {
-            JexlNodes.ensureCapacity(parentNode, fields.size() + 1);
-            JexlNode child = RebuildingVisitor.copy(orgNode);
-            parentNode.jjtAddChild(child, parentNodeChildCount);
-            child.jjtSetParent(parentNode);
-            parentNodeChildCount++;
+            JexlNode child = copy(original);
+            children.add(child);
             // remove this entry from the fieldsToValues to avoid duplication
-            for (String identifier : JexlASTHelper.getIdentifierNames(orgNode)) {
-                for (Object value : JexlASTHelper.getLiteralValues(orgNode)) {
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
+                for (Object value : JexlASTHelper.getLiteralValues(original)) {
                     fieldsToValues.remove(identifier, value);
                 }
             }
-        } else {
-            JexlNodes.ensureCapacity(parentNode, fields.size());
         }
         
         for (String field : fields) {
-            ValueSet valuesForField = fieldsToValues.get(field);
+            JexlNode child = createNodeFromValuesForField(field, fieldsToValues, original, containerType, expandValues, isNegated);
+            if (child != null) {
```

Comment:
Could we instead have a unit test for `JexlNodeFactory.createNodeFromValuesForField()` to verify that it always returns a node with a valid lineage (and fix it if it doesn't) rather than making this check in runtime here?

Reply:
Removing the lineage validation, that was leftover from testing. There is also a case where a null child would be returned, if the inputs are bad. I can work on improving code coverage.